packages/calcite-components/src/components/icon/icon.scss

```diff
@@ -10,22 +10,29 @@
   @apply text-color-icon inline-flex;
 }
 
+$icon-size-s: 16px;
+$icon-size-m: 24px;
+$icon-size-l: 32px;
+
 :host([scale="s"]) {
-  @apply h-4 w-4;
-  min-inline-size: theme("spacing.4");
-  min-block-size: theme("spacing.4");
+  inline-size: $icon-size-s;
```

Comment:
Not sure If I need the min sizes as well but the previous code had height and min size.

Reply:
Min sizes are superfluous when width and height are pixels.

Reply:
I think this prevents icon from squishing in cases where it doesn’t have room but may need to test that….

Reply:
Most icon patterns I've seen set the width and height of the icon to 100% of its container and then size the container. In that case min- sizing would be important to prevent what you're describing but if width and height are explicit then min- sizing doesn't change anything

Reply:
I'll leave it in so it doesn't break anything.